### ISSUE TITLE
[1.3.rc-4]Meteor reload endless loop

### ISSUE THREAD

**Issue by newsiberian** [meteor/meteor#6556]
Hello, since `1.3.rc-3` I started to see endless Meteor reload while changing code. Also, this happens with the latest version. Bug occurs now 100% of times but in a half of cases. Not sure, but usually this happens when I work with client-side code (maybe not). Visually it looks like:
![meteor_loop](https://cloud.githubusercontent.com/assets/11130487/13944275/a6b41cd4-f036-11e5-9d96-312b5226e21d.gif)
And you see this line all the time.

repro repo: [here](https://github.com/newsiberian/meteor-reload-loop-repro)
To repeat, try to change something in `components` within `packages/reaction-react` package several times.

**Comment by benjamn** [comment]
If you stop the server and restart it, does it start up successfully?

**Comment by newsiberian** [comment]
@benjamn, hello, yes, I do it all the time. Stop meteor, then start again. It could happen several times in 5 minutes. Maybe you could repeat it if save changes often, i.e. you could try to do several change one-by-one and save file after every change.

**Comment by benjamn** [comment]
The way I was able to reproduce this was by running `meteor test-packages`, waiting for the server to start, changing a file, then changing the same file again before the rebuild completed. Something about checking the `mtime` in `watch.isUpToDate` just wasn't reliable enough to handle multiple changes like that, so I reverted my changes to that code, so now we're back to a version that's a little slower but has been around through multiple Meteor releases. I can't reproduce the problem anymore.
